examples/gnrc_networking-subnets/setup_taps.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+PREFIX=2001:db8::/60
+
+RIOTBASE=../..
+RIOTTOOLS=$RIOTBASE/dist/tools
+TOPOLOGY=tapology.txt
+
+setup() {
+	# add address to loopback so we can test reaching an
+	# address outside the test net
+	sudo ip a a fd00:dead:beef::1/128 dev lo
+
+	echo "creating tap interfaces"
+	i=0
+	while read -r level num; do
+		$(printf -- "sudo %s/tapsetup/tapsetup -b br%s -t tap_%s -c %s\n" $RIOTTOOLS $i $level $num) > /dev/null;
```

Comment:
I could be mistaken, but I think your script (and the overall network logic) could be greatly simplified by using `veth` for the downstream connections instead and only using the bridge for the upstream most node (I don't even think you need a bridge for that tbh). See https://developers.redhat.com/blog/2018/10/22/introduction-to-linux-interfaces-for-virtual-networking#veth for an overview over the various virtual network device types in Linux.

Reply:
will this still work with `netdev_tap`?

Reply:
If the veth'd interfaces are tap interfaces, it should.

Reply:
Basically, at the moment you put a switch between every one of your nodes. With veth you connect them only with a cable.

Reply:
Well actually a switch is what I want - there is a routing node on each level, but it can also have multiple leaf nodes (which are the whole point of this exercise after all :wink:)

Now if I can use `veth` like a hub / bus and connect multiple nodes, that would be even better, but it looks like they always come in pairs.